modules/manifold/src/selectors/adaptors.js

```diff
@@ -1,12 +1,37 @@
 import {createSelector} from 'reselect';
 import {dotRange} from '@mlvis/mlvis-common/utils';
-import {getDivergenceThreshold, getMetric} from './base';
+import {
+  getNClusters,
+  getSegmentGroups,
+  getSegmentFilters,
+  getDivergenceThreshold,
+  getMetric,
+  getIsManualSegmentation,
+} from './base';
 import {getModelsMeta, getFeaturesMeta} from './compute';
 import {getModelsPerformance, getFeaturesDistribution} from './data';
 // ------------------------------------------------------------------------------------------- //
 // -- THE ADAPTOR SELECTORS DO NECESSARY TRANSFORMATION TO THE OUTPUTS OF THE DATA SELECTOR -- //
 // ------------------------------------------------------------------------------------------- //
 
+export const getSegmentIds = createSelector(
+  [getIsManualSegmentation, getNClusters, getSegmentFilters],
+  (isManual, nClusters, segmentFilters) => {
+    const nSegments = isManual ? segmentFilters.length : nClusters;
+    return Array.from(Array(nSegments).keys());
+  }
+);
+
+export const getSegmentOrdering = createSelector(
+  [getSegmentIds, getSegmentGroups],
+  (segmentIds, segmentGroups) => {
+    const nonGroup = segmentIds.filter(
+      e => !segmentGroups[0].includes(e) && !segmentGroups[1].includes(e)
+    );
+    return nonGroup.concat(segmentGroups[1]).concat(segmentGroups[0]);
```

Comment:
shall we comment on the order here?
[nonGroup, segmentGroups[1], segmentGroups[0]]
why `1` and then `0`?

Reply:
![image](https://user-images.githubusercontent.com/4994677/64985979-f5070b00-d87a-11e9-943f-ca16115f4369.png)
groups are ordered reversely, we assume user cares most about segmentGroups[0], then segmentGroups[1], then nonGroup.